### PR TITLE
feat: custom exchange [LIVE-10301]

### PR DESCRIPTION
**Pull request by Justkant**
### 📝 Description

Custom exchange module and handlers for wallet-api

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/LIVE-10301 and https://ledgerhq.atlassian.net/browse/LIVE-10281

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** There is no good way for now to test the exchange code automatically
- [ ] **Impact of the changes:**
  - Should not impact current features but might be good to make sure we don't break anything in PTX context (like buy-sell)

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by sprohaszka-ledger — COMMENTED** [review]
Don't we miss the LLM file equivalent of LLD/WebPTXPlayer/index.ts?

Otherwise, I suppose it is a copy/paste of the current code of exchange methods to a dedicated customHandler

**Comment by Justkant** [comment]
@sprohaszka-ledger Indeed I didn't do the LLM part, I'll add that ASAP

File: libs/ledger-live-common/src/wallet-api/Exchange/client.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
+import { CustomModule, Transaction, serializeTransaction } from "@ledgerhq/wallet-api-client";
```
[the remark this answers is not in the document]
Comment (sprohaszka-ledger) [reply libs/ledger-live-common/src/wallet-api/Exchange/client.ts:1]: Then it is up to PTX team, but we could totally embrace the use of the ExchangeSDK (or rename it) to be a PTX SDK (or create a dedicated PTX SDK on which the ExchangeSDK would rely).
I am not 100% confident with `tree shaking` mechanism as it relies on having a very clean dependency code (it reminds some ProGuard issue).

File: apps/ledger-live-mobile/src/components/WebPTXPlayer/index.tsx [hunk 2]
```diff
@@ -112,6 +119,96 @@ function HeaderRight({ onClose }: { onClose?: () => void }) {
   return <NavigationHeaderCloseButtonAdvanced onClose={onClose} color={colors.neutral.c100} />;
 }
 
+function usePTXCustomHandlers(manifest: Props["manifest"]) {
```
Comment (sarneijim) [inline apps/ledger-live-mobile/src/components/WebPTXPlayer/index.tsx:122]: I do not think we will be using this player (history)

**Review by Wozacosta — DISMISSED** [review]
GJ !

**Review by kallen-ledger — APPROVED** [review]
LGTM.

**Review by kallen-ledger — DISMISSED** [review]

**Review by ComradeAERGO — DISMISSED** [review]
LGTM, Great job

File: libs/exchange-module/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      files: ["src/**/*.test.{ts,tsx}"],
+      env: {
+        "jest/globals": true,
+      },
+      plugins: ["jest"],
+    },
+  ],
+  rules: {
+    "no-console": ["error", { allow: ["warn", "error"] }],
+    "@typescript-eslint/no-empty-function": "off",
```
Comment (ComradeAERGO) [inline libs/exchange-module/.eslintrc.js:17]: [no action] Out of curiosity, why do we need this one?

**Review by sprohaszka-ledger — DISMISSED** [review]
LGTM

File: libs/exchange-module/.eslintrc.js (hunk 3 above)
Comment (Justkant) [reply libs/exchange-module/.eslintrc.js:17]: We don't, it's just a copy of another module

**Review by kallen-ledger — APPROVED** [review]

**Review by Wozacosta — APPROVED** [review]

**Pull request merged**